### PR TITLE
moving aws_metadata and config_loader to this repo

### PR DESCRIPTION
**Pull request by bleachbyte**
Due to depedencies within our internal cookbooks on the `ConfigLoader` class, this moves the `aws_metadata` and `config_loader` cookbooks over to `identity-cookbooks`.

### CODE REVIEW

File: config_loader/metadata.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+maintainer 'The Authors'
+maintainer_email 'you@example.com'
+license 'All Rights Reserved'
+description 'Installs/Configures config_loader'
+long_description 'Installs/Configures config_loader'
+version '0.2.2'
+chef_version '>= 12.15.19' if respond_to?(:chef_version)
+
+depends 'citadel_fork'
+depends 'aws_metadata'
```
Comment (brodygov) [inline config_loader/metadata.rb]: Just a note to ourselves: the caller of this cookbook will have to ensure that these are both in the toplevel Berksfile.

File: config_loader/Berksfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#    https://github.com/poise/citadel/pull/35 and
+#    https://github.com/poise/citadel/pull/36.
+# 2. There is an extra build step required, and the citadel repository itself is
+#    a valid cookbook.  See the citadel-build repository for more details.  This
+#    worked before without this extra repo because we were pulling the built
+#    citadel from https://supermarket.chef.io.
+#
+# Once the pull requests are merged and a new citadel version is released we can
+# get rid of this.
+cookbook 'citadel_fork', '~> 9.1.1', git: 'https://github.com/18f/citadel-build.git'
```
Comment (brodygov) [inline config_loader/Berksfile]: We might want to add some documentation that callers need to put all the recursive dependencies in their toplevel Berksfile, since berkshelf won't process the Berksfile of dependency cookbooks.

We may also want to remove the dependency on citadel at some point, which never merged my pull requests and appears to be unmaintained.

File: identity_base_config/recipes/default.rb [hunk 3]
```diff
@@ -74,7 +74,8 @@
 # download Slack hook/channel files for id-rails-console
 %w(slackwebhook slackchannel).each do |f|
   file "/etc/login.gov/keys/#{f}" do
-    content ConfigLoader.load_config(node, f)
+    content ConfigLoader.load_config_or_nil(node, f)
+    not_if { node['config_loader']['bucket_prefix'].nil? }
```
Comment (brodygov) [inline identity_base_config/recipes/default.rb]: Why the change to `_or_nil()`? I'd think it should be an error if the webhook is missing.
I'm also not sure why the `not_if` block is needed.

Comment (bleachbyte) [reply identity_base_config/recipes/default.rb]: This is an issue that the AMI build has. When it gets to this block, the objects it attempts to load (via `load_config`) begin with `_default` instead of a URL, and since that doesn't exist (and isn't `nil`), it errors out and the Chef AMI build fails.

File: config_loader/Berksfile (hunk 2 above)
Comment (bleachbyte) [reply config_loader/Berksfile]: I'll copy/move over the comments from the previous location of these cookbooks.

File: identity_base_config/recipes/default.rb (hunk 3 above)
Comment (brodygov) [reply identity_base_config/recipes/default.rb]: Oh, I understand now. I wasn't really thinking about the context previously.

None of this delivery of secrets should be in this repo at all.

The base image delivers static files that don't differ between environments, such as the content of scripts like `id-rails-console.`

For delivering secrets like `slackwebhook`, that can't happen in the base image, because the values differ between environments, and are sensitive and shouldn't be baked into the AMI.

So this ConfigLoader.load_config() call should be happening in a recipe in identity-devops instead.

That's why it's especially good practice to separate out the static script content from the delivery of the secrets.

Comment (brodygov) [reply identity_base_config/recipes/default.rb]: (That also means that we don't need the config_loader or aws_metadata cookbooks in this repo, although it's not harmful to move them.)

File: identity_base_config/metadata.rb [hunk 4]
```diff
@@ -4,10 +4,11 @@
...
 long_description 'Identity base config and packages'
-version '0.1.3'
+version '0.1.4'
 chef_version '>= 13.0' if respond_to?(:chef_version)
 
 issues_url 'https://github.com/18F/identity-cookbooks/issues'
 source_url 'https://github.com/18F/identity-cookbooks'
 
 depends 'identity_shared_attributes'
+depends 'config_loader'
```
Comment (brodygov) [inline identity_base_config/metadata.rb]: Is this depends still true?

Comment (bleachbyte) [reply identity_base_config/metadata.rb]: Good catch; will remove and re-test.
